ci-operator/step-registry/ipi/conf/inframachineset/ipi-conf-inframachineset-commands.sh

```diff
@@ -0,0 +1,134 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if test ! -f "${KUBECONFIG}"
+then
+	echo "No kubeconfig, can not continue."
+	exit 0
+fi
+
+# Download jq
+mkdir /tmp/bin
+export PATH=$PATH:/tmp/bin
+curl -sL https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 > /tmp/bin/jq
+chmod ug+x /tmp/bin/jq
+
+# Create infra machineconfigpool
+oc create -f - <<EOF
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: infra
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,infra]}
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/infra: ""
+EOF
+
+# Get machineset name to generate a generic template
+ref_machineset_name=$(oc get machinesets -n openshift-machine-api |grep worker |awk 'NR==1{print $1}')
+
+# Get machineset temaplte
+template_machineset=$(oc get machineset $ref_machineset_name -n openshift-machine-api -o json)
+
+# Replace machine name worker to infra
+infra_machineset_name=$(echo ${template_machineset} |jq -c '.metadata.name' |sed -e 's/worker/infra/' |sed -e 's/^"//' -e 's/"$//')
+
+export ref_machineset_name infra_machineset_name
+# Get a templated json from worker machineset, change machine type and machine name
+# and pass it to oc to create a new machine set
+oc get -nopenshift-machine-api machineset $ref_machineset_name -o json \
+    | jq --arg INFRA_NODE_TYPE "${INFRA_NODE_TYPE}" '.spec.template.spec.providerSpec.value.instanceType = $INFRA_NODE_TYPE' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.metadata.name = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.template.metadata.labels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.selector.matchLabels."machine.openshift.io/cluster-api-machineset" = $infra_machineset_name' \
+    | jq --arg infra_machineset_name "${infra_machineset_name}" '.spec.template.spec.metadata.labels."node-role.kubernetes.io/infra" = ""' \
+    | jq 'del(.status)|del(.metadata.selfLink)|del(.metadata.uid)' \
+    | oc create -f -
```

Comment:
should be no secrets here.  Maybe dump to a file in `/tmp/...`, `cat` the file for easier debugging, and then `oc create -f ...` the resource?  I'm ok with your direct creation if you feel that debugging via the logged manifest will be rare enough to not be worth the trouble.